### ISSUE TITLE
Improve table component guidance, accessibility requirements, and editorial experience

### ISSUE THREAD

**Issue by oksana-c** [department-of-veterans-affairs/va.gov-cms#2932]
## Background

Some useful snippets from a [Slack thread from September 22, 2021](https://dsva.slack.com/archives/C0NGDDXME/p1632323709131900)

@AngelaFowler82 
> The important thing to know about tables from an accessibility perspective is they need to have proper headers. Also, the more straightforward the better. Tables nested within tables and tables whose individual rows and columns are of different lengths need to be avoided. 
 
> Table should really only be used when there’s a logical relationship between the elements of data, and there’s more than two. For example, a bank statement with the date of transaction, amount of transaction, and who the transaction is with makes sense to be in table form.

@joshkimux 
> If the content can be designed as a description list (e.g. 2 column definition and description), it should use a description list.
> If data/info comparison is the goal for using a table, alternative content strategies should be considered first. e.g. Can we anticipate what the user will want to compare or simplify that the comparison process through a step by step flow?
> One column lists will generally be more accessible on mobile screens since they won't (a) require a horizontal scroll or (b) create narrow, unreadable columns

We had a call Sept 30 2021 with members of sitewide content team and accessibility team. Some notes from @joshkimux 

**Main takeaways:**

- We recommend providing guidance to editors to consider thinking from a mobile first layout before a desktop layout.
- Non-matrix tables with only one header type (either column or row) should be written as lists with headings.
- For example, the tables on survivor dlc rates should be lists. See the first screenshot for an example of headings. I used headingsmap to display the order.
- Matrix tables with both column and row headers should be written as tables.
- For example, the special monthly compensation rates tables should remain as tables as it's easier for screen reader users to parse through. They will first identify the row relevant to them, and then navigate right to determine the values.

**Additional notes:**

- Zebra stripes help users visually group rows that change into blocks as users rotate screens or otherwise change context. This may be helpful particularly to folks who use screen magnification tools.
- Headings on the special monthly compensation rates page should be given an accessibility review. Accordions may need to be listed as headings for them to be discoverable to screen reader users. They are also caption elements that look like headings, but aren't.

### User Story or Problem Statement

As a content author, I need a user-friendly table UI, so I can perform my work efficiently and error-free.

As a content editor, I need guidance (and/or guardrails) about when it's appropriate to use a table vs. a list so I can feel confident that I'm creating content that will be accessible to people using screen readers.

### Affected users and stakeholders

* CMS editors

### Change management triage

The change represented by this user story will:
- [x] Be a site-wide change to appearance or key functionality (such as log-in process)
- [ ] Be a specific change that requires more than 30 minutes of work by a user to meet a publishing requirement
- [ ] Remove a piece of functionality (such as restricting editing functions, or “hardening”)
- [ ] Require action by some or all users by a specific deadline
- [ ] Change the front end and be highly visible to the public

### Hypothesis

_By improving table authoring experience, we will reduce the time required for adding/editing complex tables._

_By enhancing table field features and data validation, we will improve the usability of tables on the front-end._

_By providing better guidance, tables on VA.gov will be more useable and accessible._

**Comment by rachel-kauff** [comment]
Possible next steps: Audit which products are currently using tables so we can understand whether VAMC/field editors account for a significant amount of tables, and decide what approach to take (guidance vs. guardrails).

**Comment by ContentBrewmaster** [comment]
Added change management triage section.

**Comment by EWashb** [comment]
@laflannery do you have thoughts on this epic? This is another that seems like we might have covered with other efforts

**Comment by EWashb** [comment]
That's what I was thinking, too @laflannery. I will close the epic but keep the task as a stand-alone.
